### PR TITLE
Add Cluster Costs experimental page

### PR DESCRIPTION
**Pull request by mendaomn**
## Description

This PR adds a page reachable via the cluster actions, which shows the cost data for the selected cluster.

In order to reduce calls to Cost Explorer to a minimum, the `staleTime` of the queries is set to `Infinity`, thus new data is getting fetched only on page reload.

This PR builds upon #204 . ~~Even though it is not merged yet, this is safe to merge as it is being protected by a feature flag.~~

**Please note**: copy is not definitive and hasn't been reviewed yet

## Changes

- add new cluster action to View Costs
- add new /costs page
- set `staleTime` of queries to `Infinity`

## Screenshots

Screenshot 2023-04-28 at 16 20 03

Screenshot 2023-04-28 at 16 19 30
## How Has This Been Tested?

- manually, by mocking backend responses
- manually, by integrating locally with #204 and checking that data is graphed as expected
- unit tests

## PR Quality Checklist

- [x] I added tests to new or existing code
- [ ] I removed hardcoded strings and used [`react-i18next`](https://react.i18next.com/) library ([useTranslation hook](https://react.i18next.com/latest/usetranslation-hook) and/or [Trans component](https://react.i18next.com/latest/trans-component)), see an example [here](https://github.com/aws/aws-parallelcluster-ui/commit/a6f1e2aa46b245b5bf7500a04b83195477a5cfa5)
- [ ] I made sure no sensitive info gets logged at any time in the codebase (see [here](https://cheatsheetseries.owasp.org/cheatsheets/Logging_Cheat_Sheet.html)) (e.g. no user info or details, no stacktraces, etc.)
- [ ] I made sure that any [GitHub issue](https://github.com/aws/aws-parallelcluster-ui/issues) solved by this PR is correctly linked
- [ ] I checked that infrastructure/update_infrastructure.sh runs without any error
- [x] I checked that `npm run build` builds without any error
- [x] I checked that clusters are listed correctly
- [ ] I checked that a new cluster can be created (config is produced and dry run passes)
- [ ] I checked that login and logout work as expected

In order to increase the likelihood of your contribution being accepted, please make sure you have read both the [Contributing Guidelines](../CONTRIBUTING.md) and the [Project Guidelines](../PROJECT_GUIDELINES.md)

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: frontend/src/old-pages/Costs/CostData.tsx [hunk 1]
```diff
@@ -0,0 +1,156 @@
...
+    ],
+    [t],
+  )
+
+  const series: BarChartProps<XAxisValueType>['series'] = useMemo(
+    () =>
+      isSuccess
+        ? [
+            {
+              title: t('costMonitoring.costData.chart.title'),
```
Comment (mtfranchetto) [inline frontend/src/old-pages/Costs/CostData.tsx:96]: issue: I know this copy is provisional, but it's misleading. This is the title of the first series (we can have many), but "Spending analysis chart" should be the title of the chart itself. "Price per month" can be better IMHO, or pick a similar copy.

File: frontend/src/old-pages/Costs/CostData.tsx [hunk 2]
```diff
@@ -0,0 +1,156 @@
...
+              data: toSeriesData(months, data),
+              valueFormatter: toFullDollarAmount,
+            },
+          ]
+        : [],
+    [data, isSuccess, months, t],
+  )
+
+  const domain = consoleDomain(region)
+  const costExplorerHref = `${domain}/cost-management/home#/cost-explorer`
```
Comment (mtfranchetto) [inline frontend/src/old-pages/Costs/CostData.tsx:107]: question: have you tried the link? From what I know cost explorer is only available on `us-east-1`

Comment (mendaomn) [reply frontend/src/old-pages/Costs/CostData.tsx:107]: Yes, it works fine. Users are presented with the `Global` caption for the service region. Same for billing

File: frontend/src/old-pages/Costs/CostData.tsx (hunk 1 above)
Comment (mendaomn) [reply frontend/src/old-pages/Costs/CostData.tsx:96]: I then changed it to `Spent`, taking inspiration from Cost Explorer where they use the `Costs` name for the series. I think your point still stands though, and we might use `Monthly spending` which should play fine with the y-axis being labeled `Spent ($)`

For reference, this is how it looks like right now

image

**Review by mtfranchetto — APPROVED** [review]

**Pull request merged**